docs/basics.adoc

```diff
@@ -329,6 +329,40 @@ Note, that schema name changes may not be immediately visible in currently runni
 Should the need arise to re-define an existing schema type, it is recommended to change the name of this type to a name that is not currently (and will never be) in use. After that, a new label or key can be defined with the original name, thereby effectively replacing the old one.
 However, note that this would not affect vertices, edges, or properties previously written with the existing type. Redefining existing graph elements is not supported online and must be accomplished through a batch graph transformation.
 
+=== Schema Constraints
+
+The definition of the schema allows to configure explict property and connection constraints. These constraints allow to explicitly bind properties to a single vertex label or edge label. Moreover, the connection constraints allow to explicitly define which two vertex labels can be connected by an edge label. These constraints can be used to ensure that a graph matches a given domain model. For example for the graph of the gods, a `god` can be a brother of another `god`, but not of a `monster` and a `god` can have a property `age`, but `location` can not have a property `age`. However, these constraints are disabled by default. 
+
+Enable these schema constraints by setting `schema.constraints=true`. This setting depends on the setting `schema.default`. If config `schema.default` is set to `none`, then an `IllegalArgumentException` is thrown for schema constraint violations. If `schema.default` is not set `none`, schema constraints are automatically created, but no exception is thrown.
+
```

Comment:
Could you add a note on how these new constraints will affect an existing database in an upgrade situation?

Perhaps another important thing to mention is that the constraints aren't enforced when data is being read. In other words, if a user writes data to some property, then enables constraints and doesn't add that property to the vertex label in question, they'll still see the original property in their results. Please correct me if I'm wrong here though? Also, I should say, the intent was not to do read-side filtering so what you have is looking great, I just want to be sure the behavior is clear to the user.